runtime/context.go

```diff
@@ -41,7 +41,22 @@ var (
 	DefaultContextTimeout = 0 * time.Second
 )
 
-type rpcMethodKey struct{}
+type (
+	rpcMethodKey       struct{}
+	httpPathPatternKey struct{}
+
+	annotateOptions struct {
+		httpPath string
+	}
+
+	AnnotateOption func(ao *annotateOptions)
```

Comment:
Since we're only using this in methods that mutate the `context.Context`, maybe it would make more sense for this to be of type

```suggestion
	AnnotateOption func(ctx context.Context) context.Context
```

Then it could just add incoming/outgoing context information directly.

Reply:
Thanks!
I thiink this idea makes options simply.

Just considering if there may be usecase that not mutate context In future changes. ( e.g. add data not to ctx but to medatada.MD)

Reply:
I think we should be safe, the job of this function is to annotate the context, nothing more. Changing the metadata.MD would also require mutating the context.